acra/configuring-maintaining/tls/_index.md

```diff
@@ -2,3 +2,146 @@
 title: TLS
 bookCollapseSection: true
 ---
+
+# TLS
+
+## Common TLS flags
+
+(common for `acra-connector`, `acra-server`, `acra-translator`)
+
+* `--tls_auth=<mode>`
+
+  Set authentication mode that will be used in TLS connection with AcraConnector and database.
+
+  * `0` — do not request client certificate, ignore it if received
+  * `1` — request client certificate, but don't require it
+  * `2` — expect to receive at least one certificate to continue the handshake
+  * `3` — don't require client certificate, but validate it if client actually sent it
+  * `4` — (default) request and validate client certificate
+
+  These values correspond to [crypto.tls.ClientAuthType](https://golang.org/pkg/crypto/tls/#ClientAuthType).
+
+* `--tls_key=<filename>`
+
+  Path to private key that will be used in AcraServer's TLS handshake
+  with AcraConnector as server's key and database as client's key.
+  Empty by default.
+
+* `--tls_cert=<filename>`
+
+  Path to TLS certificate, will be sent to other peer during handshake.
+  Empty by default.
+
+* `--tls_ca=<filename>`
+
+  Path to additional CA certificate for AcraConnector and database certificate validation.
+  Empty by default.
+
+## AcraConnector specific flags
+
+* `--acraserver_tls_transport_enable={true|false}`
+
+  Enable TLS to encrypt transport between AcraConnector and AcraServer.
+  Default is `false` which means "use SecureSession instead".
+
+## AcraServer specific flags
+
+* `--acraconnector_tls_transport_enable={true|false}`
+
+  Enable TLS to encrypt transport between AcraConnector and AcraServer.
+  Default is `false` which means "use SecureSession instead".
+
+* `--tls_client_auth=<mode>`
+
+  Set authentication mode that will be used in TLS connection with AcraConnector.
+  Possible values are the same as for `--tls_auth`.
+  Default is `-1` which means "take value of `--tls_auth`".
+
+  Overrides the `--tls_auth` setting.
+
+* `--tls_client_key=<filename>`
+
+  Path to private key of the TLS certificate presented to AcraConnectors (see `--tls_client_cert`).
+  Empty by default.
+
+* `--tls_client_cert=<filename>`
+
+  Path to server TLS certificate presented to AcraConnectors (overrides `--tls_cert`).
+  Empty by default.
+
+* `--tls_client_ca=<filename>`
+
+  Path to additional CA certificate for AcraConnector certificate validation (setup if AcraConnector certificate CA is different from database certificate CA).
+  Empty by default.
+
+* `--tls_client_id_from_cert={true|false}`
+
+  Extract clientID from TLS certificate.
+  Take TLS certificate from AcraConnector's connection if `--acraconnector_tls_transport_enable` is `true`;
+  otherwise take TLS certificate from application's connection if `--acraconnector_transport_encryption_disable` is `true`.
+  Can't be used with `--tls_client_auth=0` or `--tls_auth=0`.
+  Default is `false`.
+
+* `--tls_database_auth=<mode>`
+
+  Set authentication mode that will be used in TLS connection with database.
```

Comment:
used for TLS?

Reply:
Replaced. Though I'm not sure whether there is a big difference in correctness (`used in` vs `used for`).

Reply:
Not a big, indeed. Just triggered on it from grammar perspectives